jboss-kie-wildfly-common/added/launch/jboss-kie-wildfly-security.sh

```diff
@@ -157,7 +157,7 @@ function set_application_roles_config() {
 
 function add_eap_user() {
     # If LDAP/SSO integration is enabled and only external auth is set, do not create eap users.
-    if [ "${EXTERNAL_AUTH_ONLY}" == "true" ] && ["${AUTH_LDAP_URL}x" == "x" ] && [ "${SSO_URL}x" == "x" ]; then
+    if [ "${EXTERNAL_AUTH_ONLY}" != "true" ] || ([ "${AUTH_LDAP_URL}x" == "x" ] && [ "${SSO_URL}x" == "x" ]); then
```

Comment:
@ruromero There should be added checks before for not valid options so add eap user can fail with some infromation what is wrong set. Not valid options are LDAP and SSO configured at the same time (maybe this is checked earlier) and External auth only set to true, when LDAP or SSO are not provided.
Also this check will pass if user will set EXTERNAL_AUTH_ONLY to true, but leave AUTH_LDAP_URL and SSO_URL empty.

Reply:
I think its fair to add this extra functionality but let's do it on a separate JIRA.
Regarding EXTERNAL_AUTH_ONLY = true and LDAP/SSO empty. It is correct and the user should be created because the EXTERNAL_AUTH_ONLY env var will be ignored (there is no external auth set)

Reply:
I would prefer this functionality to be part of this issue as it is related to new external property.

I do not think that ignoring EXTERNAL_AUTH_ONLY is correct, with configured property users do expect some change of behaving. If this configuration is not correct we should fail in this case. Also we should put this to property description / documentation.

Reply:
Do you mean to add a check that shows a log or that makes the script (an the pod) to fail? I am against adding more complexity to the scripts.

Regarding property description / documentation:
It is set in the templates: https://github.com/jboss-container-images/rhpam-7-openshift-image/blob/master/templates/rhpam77-prod.yaml#L339
And in the images: https://github.com/jboss-container-images/rhpam-7-openshift-image/blob/master/businesscentral/image.yaml#L89

Do you want to suggest a different description?